plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergBucketing.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg;
+
+import io.trino.spi.block.Block;
+import io.trino.spi.block.BlockBuilder;
+import io.trino.spi.type.Decimals;
+import io.trino.spi.type.TestingTypeManager;
+import io.trino.spi.type.TypeManager;
+import io.trino.spi.type.VarbinaryType;
+import org.apache.iceberg.transforms.Transform;
+import org.apache.iceberg.transforms.Transforms;
+import org.apache.iceberg.types.Type;
+import org.apache.iceberg.types.Types;
+import org.apache.iceberg.types.Types.BooleanType;
+import org.apache.iceberg.types.Types.DoubleType;
+import org.apache.iceberg.types.Types.FloatType;
+import org.apache.iceberg.types.Types.UUIDType;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.math.BigDecimal;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.function.Function;
+
+import static io.trino.plugin.iceberg.PartitionTransforms.getBucketTransform;
+import static io.trino.plugin.iceberg.TypeConverter.toTrinoType;
+import static io.trino.spi.type.Decimals.isLongDecimal;
+import static io.trino.spi.type.TypeUtils.writeNativeValue;
+import static java.lang.String.format;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatThrownBy;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNull;
+import static org.testng.Assert.assertTrue;
+
+public class TestIcebergBucketing
+{
+    private static final TypeManager TYPE_MANAGER = new TestingTypeManager();
+
+    @Test
+    public void testBucketNumberCompare()
+    {
+        // Only primitive types are covered here (logical types like timestamp, date or time maps to long)
```

Comment:
maps -> map



we should test with all logical types IMO

Reply:
Then I'd have to build a logic to map logical type to underlying storage type (like timestamp -> long) and I'll get the same result as testing long

Reply:
is it possible that some two different logical types have different bucketing behavior?
is bucketing transform defined in terms of logical or primitives?

Reply:
Bucketing is defined for logical types, but hashing operates on primitives.

Reply:
So Iceberg date is an integer, while Trino date is long. I don't think it matters, but i would feel safer if we tested all logical types here.

Reply:
Logical types byte in unexpected ways - see https://github.com/trinodb/trino/pull/8101#issuecomment-852745163.

Let's add them where possible - more coverage never hurt anyone.

Reply:
I couldn't find single place in Iceberg codebase where conversion between logical and primitive type happens. Do you have an idea how to test that?